server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -183,10 +167,35 @@ protected WritePrimaryResult<ShardUpsertRequest, ShardResponse> processRequestIt
             insertColumns,
             request.returnValues()
         );
-        if (indexer.hasUndeterministicSynthetics()) {
-            // This change also applies for RawIndexer if it's used.
-            // RawIndexer adds non-deterministic generated columns in addition to _raw and uses same request.
-            request.insertColumns(indexer.insertColumns(insertColumns));
+
+        Indexer updatingIndexer = null;
+        if (updateToInsert != null) {
+            insertColumns = updateToInsert.columns();
```

Comment:
Is it really correct/safe/needed to overwrite the `insertColumns`?

Reply:
the only reason I did that was avoiding `IndexOutOfBounds` when checking that  first ident == _raw`.

It's actually no needed and indeed confusing, pushed a change to check _raw  in a clearer way and got rid of that conversion.